docs/source/API/core/macros-special/if_on_host_or_device.rst

```diff
@@ -0,0 +1,111 @@
+.. _kokkos_if_on_host_device_macros:
+
+=================================================
+``KOKKOS_IF_ON_HOST`` and ``KOKKOS_IF_ON_DEVICE``
+=================================================
+
+.. role:: cpp(code)
+   :language: cpp
+
+Overview
+========
+The ``KOKKOS_IF_ON_HOST`` and ``KOKKOS_IF_ON_DEVICE`` macros are a pair of
+function-like macros introduced in Kokkos 3.6 that enable **portable
+conditional compilation** within a **single ``KOKKOS_FUNCTION`` body**. They
+allow you to select which code is compiled and executed based on whether the
+function is running on the host (CPU) or a device (GPU, etc.). These macros
+provide an alternative to non-portable preprocessor idioms like #ifdef
+``__CUDA_ARCH__``.
+
+Motivation
+==========
+Traditional preprocessor directives like ``#ifdef __CUDA_ARCH__`` rely on a
+split compilation model, where host and device code are compiled in separate
+passes.  This model is not supported by modern unified compilation approaches,
+such as the one used by the NVIDIA HPC compiler (NVC++). As a result, code
```

Comment:
We need to get rid of this, NVC++ does not support CUDA anymore and they are not committed to doing it. Maybe mention SYCL or OpenMPTarget instead? How does it work there?